### PR TITLE
[TT-11672] Update description of CertificatesConfig.Upstream

### PR DESCRIPTION
**Pull request by dcs3spp**
## Description
The tyk-docs repository has had a [PR](https://github.com/TykTechnologies/tyk-docs/pull/4306) raised that proposes a config description update for _CertificatesConfig.Upstream_ within the autogenerated Gateway config file from source. This PR has been created on behalf of @matthewdickinson with proposed change for review. Once reviewed and if approved, the fix would need to be synced into docs repository.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

___

## **Type**
documentation

___

## **Description**
- Updated the `CertificatesConfig.Upstream` documentation in `config/config.go` to improve clarity on how to specify upstream mutual TLS certificates globally. This includes a more detailed description and examples for both `tyk.conf` and environmental variable configurations.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>config.go</strong><dd><code>Update Documentation for Upstream Mutual TLS Certificates </code><br><code>Configuration</code></dd></summary>
<hr>

config/config.go
<li>Updated the description for <code>CertificatesConfig.Upstream</code> to clarify the <br>format and usage of upstream mutual TLS certificates.<br> <li> Provided examples for configuring <code>CertificatesConfig.Upstream</code> in <br><code>tyk.conf</code> and as an environmental variable.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6169/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by matthewdickinson** [comment]
@dcs3spp Is there anything you need me to do to move this along, or is it still waiting for reviews?

**Comment by matthewdickinson** [comment]
> @dcs3spp Is there anything you need me to do to move this along, or is it still waiting for reviews?

@dcs3spp Is there anything I can do to help correct this documentation or is that fix being handled elsewhere?

**Comment by dcs3spp** [comment]
Hi @lghiur @tbuchaillot would it be possible to review and merge PR for @matthewdickinson ?

**Review by lghiur — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -518,7 +518,9 @@ type CoProcessConfig struct {
 
 type CertificatesConfig struct {
 	API []string `json:"apis"`
-	// Specify upstream mutual TLS certificates at a global level in the following format: `{ "<host>": "<cert>" }``
+	// Specify upstream mutual TLS certificates at a global level as a map of domain -> certificate id or path
+	// In tyk.conf, this would be `"security": {"certificates": {"upstream": {"*": "/certs/default.pem", "example.com": "ab23ef123"}}},`; as an environmental variable it would be
```
Comment (lghiur) [inline config/config.go:522]: @andrei-tyk spotted that `environmental` should be `environment`

Comment (dcs3spp) [reply config/config.go]: Thanks @andrei-tyk @lghiur I have updated content to environment

Comment (dcs3spp) [reply config/config.go]: @lghiur could the PR be merged into main branch ?

Comment (lghiur) [reply config/config.go]: lgtm

File: config/config.go [hunk 2]
```diff
@@ -518,7 +518,9 @@ type CoProcessConfig struct {
 
 type CertificatesConfig struct {
 	API []string `json:"apis"`
-	// Specify upstream mutual TLS certificates at a global level in the following format: `{ "<host>": "<cert>" }``
+	// Specify upstream mutual TLS certificates at a global level as a map of domain -> certificate id or path
```
Comment (andyo-tyk) [inline config/config.go:521]:
```suggestion
	// Upstream is used to specify the certificates to be used in mutual TLS connections to upstream services. These are set at gateway level as a map of domain -> certificate id or path.
```

File: config/config.go [hunk 3]
```diff
@@ -518,7 +518,9 @@ type CoProcessConfig struct {
...
+	// In tyk.conf, this would be `"security": {"certificates": {"upstream": {"*": "/certs/default.pem", "example.com": "ab23ef123"}}},`; as an environment variable it would be
```
Comment (andyo-tyk) [inline config/config.go:522]:
```suggestion
	// For example if you want Tyk to use the certificate `ab23ef123` for requests to the `example.com` upstream and `/certs/default.pem` for all other upstreams then:
	// In `tyk.conf` you would configure `"security": {"certificates": {"upstream": {"*": "/certs/default.pem", "example.com": "ab23ef123"}}}`
```

File: config/config.go [hunk 4]
```diff
@@ -518,7 +518,9 @@ type CoProcessConfig struct {
...
+	// `*:/certs/default.pem,example.com:ab23ef123`.
```
Comment (andyo-tyk) [inline config/config.go:523]:
```suggestion
	//  And if using environment variables you would set `TYK_GATEWAY_SECURITY_CERTIFICATES_UPSTREAM` to `*:/certs/default.pem,example.com:ab23ef123`.
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
@jeffy-mathew can you please double check the env var name?

File: config/config.go (hunk 4 above)
Comment (jeffy-mathew) [reply config/config.go:523]: @andyo-tyk we are auto generating the env variable name here

Comment (andyo-tyk) [reply config/config.go:523]:
```suggestion
	// And if using environment variables you would set this to `*:/certs/default.pem,example.com:ab23ef123`.
```

**Review by andyo-tyk — APPROVED** [review]
Approved, thanks.

**Comment by dcs3spp** [comment]
@andrei, @Jeff or @titpetric how can the CI error [Release / upgrade-rpm (arm64, amazonlinux:2)](https://github.com/TykTechnologies/tyk/actions/runs/10507377125/job/29428698949?pr=6169) be resolved to enable the PR to be merged?

**Pull request merged**